runtime/compiler/control/J9Options.cpp

```diff
@@ -2041,55 +2118,74 @@ J9::Options::fePreProcess(void * base)
       if (xxJITaaSClientArgIndex >= 0 || xxJITaaSServerArgIndex >= 0)
          {
          if (xxJITaaSClientArgIndex > xxJITaaSServerArgIndex)   // client mode
+            compInfo->getPersistentInfo()->setJITaaSMode(CLIENT_MODE);
+         else                                                  // server mode
+            compInfo->getPersistentInfo()->setJITaaSMode(SERVER_MODE);
+         hasModeOption = true;
+         }
+      // If no option, check for env var TR_JITaaSServerMode, TR_JITaaSClientMode
+      if (!hasModeOption)
+         {
+         char *envVarServerMode = feGetEnv("TR_JITaaSServerMode");
+         char *envVarClientMode = feGetEnv("TR_JITaaSClientMode");
+         if (envVarServerMode != NULL && *(envVarServerMode) == '1' && envVarClientMode != NULL && *(envVarClientMode) == '1')
             {
+            compInfo->getPersistentInfo()->setJITaaSMode(NONJITaaS_MODE);
+            printf("Both env vars are set, run in non-JITaaSMode. TR_JITaaSServerMode = 1, TR_JITaaSClientMode = 1\n");
```

Comment:
Is there any functions that we are using to produce message to user, other than printf?

Reply:
Search through this file for NLS